Refresh.GameServer/Importing/AssetImporter.cs

```diff
@@ -90,8 +89,77 @@ public void ImportFromDataStore(GameDatabaseContext context, IDataStore dataStor
             AssetHash = hash,
             AssetType = this.DetermineAssetType(data, platform),
             IsPSP = platform == TokenPlatform.PSP,
+            SizeInBytes = data.Length,
         };
+        
+        if (AssetTypeHasDependencyTree(asset.AssetType, data))
+        {
+            try
+            {
+                List<string> dependencies = this.ParseDependencyTree(data);
+                foreach (string dependency in dependencies)
+                {
+                    asset.Dependencies.Add(dependency);
+                }
+            }
+            catch (Exception e)
+            {
+                this.Warn($"Could not parse dependency tree for {hash}: {e}");
+            }
+        }
 
         return asset;
     }
+
+    [Pure]
+    private static bool AssetTypeHasDependencyTree(GameAssetType type, byte[] data)
+    {
+        if (type is GameAssetType.Jpeg
+            or GameAssetType.Png
+            or GameAssetType.Tga
+            or GameAssetType.Texture
+            or GameAssetType.GameDataTexture
+            or GameAssetType.Unknown)
+        {
+            return false;
+        }
+        
+        #if DEBUG
+        char typeChar = (char)data[3];
+        if (typeChar != 'b') throw new Exception($"Asset type {type} is not binary (char was '{typeChar}')");
+        #endif
+
+        return true;
+    }
+
+    private List<string> ParseDependencyTree(byte[] data)
+    {
+        List<string> dependencies = new();
+        
+        // Parse dependency table
+        MemoryStream ms = new(data);
+        BEBinaryReader reader = new(ms);
+
+        ms.Seek(8, SeekOrigin.Begin);
+        uint dependencyTableOffset = reader.ReadUInt32();
+
+        ms.Seek(dependencyTableOffset, SeekOrigin.Begin);
+        uint dependencyCount = reader.ReadUInt32();
+
+        this.Debug($"Dependency count offset: {dependencyTableOffset}, count: {dependencyCount}");
+        for (int i = 0; i < dependencyCount; i++)
+        {
+            byte flags = reader.ReadByte();
+            if ((flags & 0x1) != 0) // UGC/SHA1
+            {
+                byte[] hashBytes = reader.ReadBytes(20);
```

Comment:
This allocates, there should be a stack allocation outside the `for` loop that you then read into using `.ReadAtLeast`